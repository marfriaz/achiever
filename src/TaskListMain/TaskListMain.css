.TaskListMain ul {
  list-style-type: none;
  margin: 0;
  padding: 10px;
}

.TaskListMain__button-container {
  max-width: 380px;
  margin: 0 10px;
}

.TaskListMain__add-task-button {
  margin: 0 auto 20px;
}
.TaskListMain__add-task-button:hover {
  background-color: #29862e;
  color: white;
}
