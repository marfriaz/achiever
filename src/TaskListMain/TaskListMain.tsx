import React, { useContext } from "react";
import { Link, useParams } from "react-router-dom";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import Task from "../Task/Task";
import CircleButton from "../CircleButton/CircleButton";
import ApiContext from "../ApiContext";
import { getTasksForGoal } from "../tasks-helpers";
import "./TaskListMain.css";

interface ParamTypes {
  goalId: string;
}

export default function TaskListMain() {
  const { goalId } = useParams<ParamTypes>();
  const { tasks } = useContext(ApiContext)!;

  const tasksForGoal = getTasksForGoal(tasks!, goalId);
  const handleDeleteTask = (taskId: number) => {};

  return (
    <section className="TaskListMain">
      <ul>
        {tasksForGoal!.map((task) => (
          <li key={task.id}>
            <Task
              id={task.id}
              name={task.name}
              modified={task.date_created}
              onDeleteTask={handleDeleteTask}
            />
          </li>
        ))}
      </ul>
      <div className="TaskListMain__button-container">
        <CircleButton
          tag={Link}
          to="/add-task"
          role="link"
          type="button"
          className="TaskListMain__add-task-button"
        >
          <FontAwesomeIcon icon="plus" />
          <br />
          Task
        </CircleButton>
      </div>
    </section>
  );
}
