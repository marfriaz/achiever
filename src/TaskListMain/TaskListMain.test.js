import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import TaskListMain from "./TaskListMain";

describe(`TaskListMain component`, () => {
  it("renders a .TaskListMain by default", () => {
    const wrapper = shallow(<TaskListMain />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  // enzyme doesn't yet support React.createContext
  it.skip("renders a Task in ul for each tasks in array", () => {
    const props = {
      match: {
        params: {
          goalId: "THIS_FOLDER_ID",
        },
      },
    };
    const context = {
      tasks: [
        {
          id: "cbc787a0-ffaf-11e8-8eb2-f2801f1b9fd1",
          name: "Dogs",
          modified: "2019-01-03T00:00:00.000Z",
          goalId: "b0715efe-ffaf-11e8-8eb2-f2801f1b9fd1",
          content: "Corporis accusamus placeat.\n \rUnde.",
        },
        {
          id: "d26e0034-ffaf-11e8-8eb2-f2801f1b9fd1",
          name: "Cats",
          modified: "2018-08-15T23:00:00.000Z",
          goalId: "THIS_FOLDER_ID",
          content: "Eos\n \rlaudantium.",
        },
        {
          id: "d26e01a6-ffaf-11e8-8eb2-f2801f1b9fd1",
          name: "Pigs",
          modified: "2018-03-01T00:00:00.000Z",
          goalId: "THIS_FOLDER_ID",
          content: "Occaecati dignissimos\nvoluptatum nihil.",
        },
        {
          id: "d26e0570-ffaf-11e8-8eb2-f2801f1b9fd1",
          name: "Birds",
          modified: "2019-01-04T00:00:00.000Z",
          goalId: "b0715efe-ffaf-11e8-8eb2-f2801f1b9fd1",
          content: "Eum culpa odit.",
        },
      ],
    };
    const ul = shallow(<TaskListMain {...props} />, context).find("ul");
    expect(toJson(ul)).toMatchSnapshot();
  });
});
