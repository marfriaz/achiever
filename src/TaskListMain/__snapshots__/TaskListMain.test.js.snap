// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskListMain component renders a .TaskListMain by default 1`] = `
<section
  className="TaskListMain"
>
  <ul />
  <div
    className="TaskListMain__button-container"
  >
    <NavCircleButton
      className="TaskListMain__add-task-button"
      tag={[Function]}
      to="/add-task"
      type="button"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon="plus"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        symbol={false}
        title=""
        transform={null}
      />
      <br />
      Task
    </NavCircleButton>
  </div>
</section>
`;

exports[`TaskListMain component renders a Task in ul for each tasks in array 1`] = `
<ul>
  <li
    key="cbc787a0-ffaf-11e8-8eb2-f2801f1b9fd1"
  >
    <Task
      id="cbc787a0-ffaf-11e8-8eb2-f2801f1b9fd1"
      modified="2019-01-03T00:00:00.000Z"
      name="Dogs"
    />
  </li>
  <li
    key="d26e0034-ffaf-11e8-8eb2-f2801f1b9fd1"
  >
    <Task
      id="d26e0034-ffaf-11e8-8eb2-f2801f1b9fd1"
      modified="2018-08-15T23:00:00.000Z"
      name="Cats"
    />
  </li>
  <li
    key="d26e01a6-ffaf-11e8-8eb2-f2801f1b9fd1"
  >
    <Task
      id="d26e01a6-ffaf-11e8-8eb2-f2801f1b9fd1"
      modified="2018-03-01T00:00:00.000Z"
      name="Pigs"
    />
  </li>
  <li
    key="d26e0570-ffaf-11e8-8eb2-f2801f1b9fd1"
  >
    <Task
      id="d26e0570-ffaf-11e8-8eb2-f2801f1b9fd1"
      modified="2019-01-04T00:00:00.000Z"
      name="Birds"
    />
  </li>
</ul>
`;
